native/src/main/java/io/ballerina/stdlib/ftp/client/FtpClientHelper.java

```diff
@@ -266,57 +244,37 @@ public void close() {
     }
 
     private static void callStreamNext(Environment env, BObject entity, BufferHolder bufferHolder,
-                                       BObject iteratorObj, CountDownLatch latch) {
-        env.getRuntime().invokeMethodAsyncConcurrently(iteratorObj, BYTE_STREAM_NEXT_FUNC, null, null, new Callback() {
-            @Override
-            public void notifySuccess(Object result) {
-                if (result == bufferHolder.getTerminalType()) {
-                    handleStreamEnd();
-                    return;
-                }
-                BArray arrayValue = ((BMap) result).getArrayValue(FIELD_VALUE);
-                if (arrayValue == null) {
-                    handleStreamEnd();
-                    return;
-                }
-                byte[] bytes = arrayValue.getBytes();
-                bufferHolder.setBuffer(bytes);
-                bufferHolder.setTerminal(false);
-                latch.countDown();
-            }
-
-            private void handleStreamEnd() {
-                entity.addNativeData(ENTITY_BYTE_STREAM, null);
-                bufferHolder.setTerminal(true);
-                latch.countDown();
+                                       BObject iteratorObj) {
+        try {
+            Object result = env.getRuntime().startIsolatedWorker(iteratorObj, BYTE_STREAM_NEXT_FUNC, null,
```

Comment:
Should we run this within a separate virtual thread? (Similar case in L271)

Reply:
No need since we need to block until result. Can we use call here?

Reply:
Tried using it before, but the tests seem to fail. I think this is expected to run concurrently. Any ideas to try and achieve the same outcome with `call()`?

Reply:
Fixed with 8e183a2c2913379d222c5e85fab694f3b63baac4